usr/src/uts/common/fs/zfs/spa.c

```diff
@@ -5471,6 +5472,7 @@ spa_vdev_remove(spa_t *spa, uint64_t guid, boolean_t unspare)
 		    ZPOOL_CONFIG_L2CACHE, l2cache, nl2cache, nv);
 		spa_load_l2cache(spa);
 		spa->spa_l2cache.sav_sync = B_TRUE;
+		spa_event_notify(spa, vd, ESC_ZFS_VDEV_REMOVE_AUX);
```

Comment:
I also realized that vd would be NULL, but I decided to pass vd instead of NULL for consistency's sake.  Passing NULL instead would be fine too; it's just a matter of style as far as I'm concerned.